fvm/evm/stdlib/contract.cdc

```diff
@@ -288,6 +290,59 @@ contract EVM {
                 value: value.attoflow
             ) as! Result
         }
+
+        /// Bridges the given NFT to the EVM environment, requiring a Provider from which to withdraw a fee to fulfill
+        /// the bridge request
+        access(all)
+        fun depositNFT(
+            nft: @NonFungibleToken.NFT,
+            feeProvider: &{FungibleToken.Provider}
+        ) {
+            EVM.borrowBridgeAccessor().depositNFT(nft: <-nft, to: self.address(), feeProvider: feeProvider)
+        }
+
+        /// Bridges the given NFT from the EVM environment, requiring a Provider from which to withdraw a fee to fulfill
+        /// the bridge request. Note: the caller should own the requested NFT in EVM
+        access(all)
+        fun withdrawNFT(
+            type: Type,
+            id: UInt256,
+            feeProvider: &{FungibleToken.Provider}
+        ): @NonFungibleToken.NFT {
+            return <- EVM.borrowBridgeAccessor().withdrawNFT(
+                caller: &self as &CadenceOwnedAccount,
+                type: type,
+                id: id,
+                feeProvider: feeProvider
+            )
+        }
+
+        /// Bridges the given Vault to the EVM environment, requiring a Provider from which to withdraw a fee to fulfill
+        /// the bridge request
+        access(all)
+        fun depositTokens(
+            vault: @FungibleToken.Vault,
+            feeProvider: &{FungibleToken.Provider}
+        ) {
+            EVM.borrowBridgeAccessor().depositTokens(vault: <-vault, to: self.address(), feeProvider: feeProvider)
```

Comment:
I have difficulty understanding why we pass self.address() for to here. but we are going to talk about it

Reply:
The `to` value here tells the bridge the EVMAddress to which transfer the bridged asset. The idea is that depositing an NFT/FT to a COA would deposit the EVM representation of the asset to the caller on the EVM side. While we could enable a caller to define that themselves and basically execute a bridge + transfer, feedback pointed to this being a/ the more common case and b/ simpler & easier to grok.

For those that do wish to define a distinct address, they can call the bridge contract directly - see [here](https://github.com/onflow/flow-evm-bridge/blob/main/cadence/contracts/bridge/FlowEVMBridge.cdc#L189L194)

Reply:
right the self here refer to the COA.